gcp_variant_transforms/beam_io/vcf_parser.py

```diff
@@ -646,14 +434,19 @@ def _init_parent_process(self, pysam_read, variants_write):
     self._vcf_reader = libcbcf.VariantFile(into_pysam, 'r')
     self._original_info_list = self._vcf_reader.header.info.keys()
 
-  def _init_child_process(self, variants_read, pysam_write, header_lines):
+  def _init_child_process(
+      self, variants_read, pysam_write, header_lines, pre_infer_headers):
     # Child process' task is to populate data into the pipe that feeds
     # VariantFile class - first by populating all of the header lines, and then
     # by redirecting the data received from the second pipe.
 
     # Write Header Lines into PySam.
     to_pysam_pipe = os.fdopen(pysam_write, 'w')
-    to_pysam_pipe.write('\n'.join(header_lines) + '\n')
+    if pre_infer_headers:
+      to_pysam_pipe.write(header_lines[0] + '\n') # fileformat line
+      to_pysam_pipe.write(header_lines[-1] + '\n') # `#CHROM...` line
```

Comment:
If I understood your comment correctly, I have to disagre, since I don't think we can use them as consts - first line defines the format and last line is CHROM line, that specifies samples.

The existence of first line is forced in this file, since if we cannot find it, we add it ourselves. The existence of last line is also required as otherwise it would break pysam anyways - it is checked in vcf_header_io in the verify_header() method (that was removed from here in this PR).

Please correct if I misunderstood you.